### DISCUSSION TITLE
How to check mocked function is called?

### DISCUSSION THREAD

**Discussion by steven-twerdochlib** [vitest-dev/vitest#5547] [Q&A] [answered]
Hi,
I have the following mock:
`vi.mock('@/api', async () => {
  const mod = await vi.importActual('@/api');
  return {
    ...mod,
    apolloMutation: () => {
      return Promise.resolve();
    },
    apolloQuery: () => {
      return Promise.resolve();
    },
    axiosGet: vi.fn(),
    default: mod
  }
});`
And I want to check that apolloQuery has been called for example. How would I do this?
Another example is:
`const wrapper = shallowMount(RandomFileName, {
      global: {
        plugins: [store],
        mocks: {
          $router: {
            push: vi.fn()
          },
          isAvailable: true
        },
      },
      propsData,
      data() {
        return {};
      }
    });`
How to check router.push() was called?

**Comment by hi-ogawa** [comment, marked as answer]
> And I want to check that apolloQuery has been called for example. How would I do this?

It could be something like this:

```diff,ts
- apolloQuery: () => { return Promise.resolve(); }
+ apolloQuery: vi.fn(() => { return Promise.resolve(); })
```

```ts,diff
import { apolloQuery } from "@/api";

vi.mock("@/api", ...);

it("...", () => {
  // do something with apolloQuery ...

  // then assert
  expect(apolloQuery).toHaveBeenCalled()  
})
```
